### PR TITLE
Split models module

### PR DESCRIPTION
**Pull request by adamcik**
Also sneaks in explicit URI and date field type placeholders.

### CODE REVIEW

File: mopidy/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
+from __future__ import absolute_import, unicode_literals
+
+from mopidy.models import fields
+from mopidy.models.immutable import ImmutableObject
+
+# TODO: remove the following "exports" once users have been migrated
+from mopidy.models.serialize import model_json_decoder, ModelJSONEncoder  # noqa
```
Comment (jodal) [inline mopidy/models/__init__.py]: Add an `__all__` instead of noqa-ing this?

File: mopidy/models/fields.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+class Integer(Field):
+
+    """
+    :class:`Field` for storing integer numbers.
+
+    :param default: default value for field
+    :param min: field value must be larger or equal to this value when set
+    :param max: field value must be smaller or equal to this value when set
+        """
```
Comment (jodal) [inline mopidy/models/fields.py]: Odd indentation

File: mopidy/models/fields.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+class Date(String):
+    pass  # TODO: make this check for YYYY-MM-DD, YYYY-MM, YYYY using stftime.
+
+
+class Identifier(String):
+    def validate(self, value):
+        return intern(str(super(Identifier, self).validate(value)))
+
+
+class Uri(Identifier):
```
Comment (jodal) [inline mopidy/models/fields.py]: "class URI"?

File: mopidy/models/fields.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    def __init__(self, default=None):
+        # TODO: normalize to unicode?
+        # TODO: only allow unicode?
+        # TODO: disallow empty strings?
+        super(String, self).__init__(type=basestring, default=default)
+
+
+class Date(String):
+    pass  # TODO: make this check for YYYY-MM-DD, YYYY-MM, YYYY using stftime.
```
Comment (jodal) [inline mopidy/models/fields.py]: s/stftime/strftime/

**Comment by jodal** [comment]
+1 with comments

**Comment by adamcik** [comment]
Done.

**Pull request merged**